docs/proposals/nap-waf.md

```diff
@@ -0,0 +1,753 @@
+    classDef cloudAuth fill:#e8f5e8,stroke:#228b22,stroke-width:2px
+
+    class SecTeam,Compiler,Store external
+    class NGFPod control
+    class NGFServiceAccount cloudAuth
+    class Gateway,HTTPRoute gateway
+    class WafEnforcer,WafConfigMgr,PolicyVol,NginxProxy,WafPolicy wafRequired
+    class Application app
+    class PolicyVol,ConfigVol volume
+    class PublicEndpoint endpoint
+    class Secret optional
+
+    %% Notes about the architecture
+    classDef note fill:#fffacd,stroke:#daa520,stroke-width:1px,stroke-dasharray: 5 5
+    Note1[📝 Note: Architecture supports multiple Gateways<br/>with selective WAF enablement per Gateway]
+    Note2[📝 Note: Secret only required for fallback authentication<br/>Native cloud auth uses NGF Service Account annotations]
+    class Note1,Note2 note
+```
+
+This architecture diagram illustrates the complete NAP v5 WAF integration within NGINX Gateway Fabric, showing the flow from policy development to traffic protection:
+
+**External Policy Management (Blue):** Security teams develop WAF policies using NAP v5 JSON schema, compile them using NAP v5 compiler tools, and publish the compiled policy bundles to accessible storage locations (S3, HTTP servers, or MinIO).
+
+**Control Plane (Purple):** The NGF Pod in the `nginx-gateway` namespace acts as the centralized control plane, watching for NginxProxy and WafPolicy resources across application namespaces, fetching compiled policies from external storage using appropriate authentication, and distributing policy configurations to NGINX Pods via secure gRPC connections.
+
+**Data Plane (Green/Red):** When WAF is enabled through NginxProxy configuration, each Gateway deploys as a multi-container NGINX Pod containing the main NGINX container with NAP module, plus the required WAF Enforcer and WAF Config Manager containers. These containers communicate through shared ephemeral volumes rather than network calls, maintaining NAP v5's architectural requirements.
+
+**Application Namespace Resources:** All user-facing resources (Gateway, HTTPRoute, WafPolicy, NginxProxy, and optional authentication Secret) reside in application namespaces for proper isolation and RBAC management. The Secret is only required when using fallback authentication methods - native cloud authentication uses annotations on the NGF service account in the nginx-gateway namespace.
+
+**Traffic Flow (Yellow/Gold):** Client traffic flows through the public load balancer endpoint to the WAF-protected NGINX container, where NAP v5 applies security policies before forwarding clean traffic to backend applications.
+
+**Resource Relationships:** WafPolicy resources attach to HTTPRoutes via Gateway API ExtensionRef filters, while NginxProxy resources enable WAF functionality at the Gateway level. This design supports selective WAF deployment where some Gateways have WAF protection while others remain standard, all managed by a single NGF control plane.
```

Comment:
Is WafPolicy going to be a Filter or a Policy? Filter would apply per rule (or request), Policy would apply per-HTTPRoute (or hostname(s), essentially). What about attaching the Policy at the Gateway level, to apply to all Routes? Do we need the granularity or is this meant to be more global for all traffic in the nginx instance?

Sidenote: will we support GRPCRoute?

Reply:
Ultimately, where do the directives get set in nginx?